openvino-rs 0.7.1 is yanked, new version has breaking changes

I found this weird issue after `cargo clean`, turns out 0.7.1 which we use is yanked (taken out altogether?).

The next version has a different API call to create Tensors, therefore `object_detection` is broken as of now (at least for non-nao builds).

- Release notes: https://github.com/intel/openvino-rs/releases/tag/v0.7.2
- The API change: https://github.com/intel/openvino-rs/commit/b6eacef9b5ab2cc964abbf02be3c74b0d7a948b1

```
error[E0599]: no function or associated item named `new_from_host_ptr` found for struct `Tensor` in the current scope
   --> crates/object_detection/src/pose_detection.rs:155:30
    |
155 |         let tensor = Tensor::new_from_host_ptr(
    |                              ^^^^^^^^^^^^^^^^^ function or associated item not found in `Tensor`
    |
note: if you're trying to build a new `Tensor` consider using one of the following associated functions:
      Tensor::new
      Tensor::from_ptr
   --> /home/darshana/.cargo/registry/src/index.crates.io-6f17d22bba15001f/openvino-0.7.2/src/tensor.rs:21:5
    |
21  |     pub fn new(element_type: ElementType, shape: &Shape) -> Result<Self> {
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
...
33  |     pub(crate) fn from_ptr(ptr: *mut ov_tensor_t) -> Self {
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -138,7 +138,7 @@ nix = { version = "0.28", features = ["ioctl"] }
 num-derive = "0.4.2"
 num-traits = "0.2"
 once_cell = "1.19.0"
-openvino = { version = "0.7.1", features = ["runtime-linking"] }
+openvino = { version = "0.7.2", features = ["runtime-linking"] }
 opn = { path = "crates/opn" }
 object_detection = { path = "crates/object_detection" }
 opusfile-ng = "0.1.0"

--- a/crates/object_detection/src/pose_detection.rs
+++ b/crates/object_detection/src/pose_detection.rs
@@ -254,17 +254,3 @@ fn non_maximum_suppression(
 
     poses
 }
-
-// Allowing dead code for future use of `scratchpad` once it is restored.
-#[allow(dead_code)]
-trait AsBytes {
-    fn as_bytes(&self) -> &[u8];
-}
-
-impl AsBytes for [f32] {
-    fn as_bytes(&self) -> &[u8] {
-        unsafe {
-            std::slice::from_raw_parts(self.as_ptr() as *const u8, std::mem::size_of_val(self))
-        }
-    }
-}

--- a/crates/object_detection/src/pose_detection.rs
+++ b/crates/object_detection/src/pose_detection.rs
@@ -152,11 +152,11 @@ impl PoseDetection {
         }
 
         let mut infer_request = self.network.create_infer_request()?;
-        let tensor = Tensor::new_from_host_ptr(
-            ElementType::F32,
-            &self.network.get_input()?.get_shape()?,
-            self.scratchpad.as_bytes(),
-        )?;
+        let mut tensor = Tensor::new(ElementType::F32, &self.network.get_input()?.get_shape()?)?;
+        tensor
+            .buffer_mut()?
+            .copy_from_slice(self.scratchpad.as_bytes());
+
         infer_request.set_input_tensor(&tensor)?;
 
         {

--- a/crates/object_detection/src/pose_detection.rs
+++ b/crates/object_detection/src/pose_detection.rs
@@ -38,8 +38,6 @@ const STRIDE: usize = DETECTION_IMAGE_HEIGHT * DETECTION_IMAGE_WIDTH;
 #[derive(Deserialize, Serialize)]
 pub struct PoseDetection {
     #[serde(skip, default = "deserialize_not_implemented")]
-    scratchpad: Box<[f32]>,
-    #[serde(skip, default = "deserialize_not_implemented")]
     network: CompiledModel,
 }
 
@@ -108,16 +106,7 @@ impl PoseDetection {
             bail!("expected exactly one input and one output");
         }
 
-        let input_shape = network
-            .get_input_by_index(0)
-            .wrap_err("failed to get input node")?
-            .get_shape()
-            .wrap_err("failed to get shape of input node")?;
-        let number_of_elements = input_shape.get_dimensions().iter().product::<i64>();
-        let scratchpad = vec![0.0; number_of_elements as usize].into_boxed_slice();
-
         Ok(Self {
-            scratchpad,
             network: core.compile_model(&network, DeviceType::CPU)?,
         })
     }
@@ -139,10 +128,12 @@ impl PoseDetection {
         };
 
         let image = context.image;
+
+        let mut tensor = Tensor::new(ElementType::F32, &self.network.get_input()?.get_shape()?)?;
         {
             let earlier = SystemTime::now();
 
-            load_into_scratchpad(self.scratchpad.as_mut(), image);
+            load_into_scratchpad(tensor.get_data_mut::<f32>()?, image);
 
             context.preprocess_duration.fill_if_subscribed(|| {
                 SystemTime::now()
@@ -152,11 +143,6 @@ impl PoseDetection {
         }
 
         let mut infer_request = self.network.create_infer_request()?;
-        let mut tensor = Tensor::new(ElementType::F32, &self.network.get_input()?.get_shape()?)?;
-
-        tensor
-            .get_raw_data_mut()?
-            .copy_from_slice(self.scratchpad.as_bytes());
 
         infer_request.set_input_tensor(&tensor)?;
 
@@ -269,6 +255,8 @@ fn non_maximum_suppression(
     poses
 }
 
+// Allowing dead code for future use of `scratchpad` once it is restored.
+#[allow(dead_code)]
 trait AsBytes {
     fn as_bytes(&self) -> &[u8];
 }

--- a/crates/object_detection/src/pose_detection.rs
+++ b/crates/object_detection/src/pose_detection.rs
@@ -153,8 +153,9 @@ impl PoseDetection {
 
         let mut infer_request = self.network.create_infer_request()?;
         let mut tensor = Tensor::new(ElementType::F32, &self.network.get_input()?.get_shape()?)?;
+
         tensor
-            .buffer_mut()?
+            .get_raw_data_mut()?
             .copy_from_slice(self.scratchpad.as_bytes());
 
         infer_request.set_input_tensor(&tensor)?;
@@ -169,7 +170,7 @@ impl PoseDetection {
                     .expect("time ran backwards")
             });
         }
-        let mut prediction = infer_request.get_output_tensor()?;
+        let prediction = infer_request.get_output_tensor()?;
         let prediction =
             ArrayView::from_shape((56, MAX_DETECTIONS), prediction.get_data::<f32>()?)?;
 

--- a/crates/object_detection/src/pose_detection.rs
+++ b/crates/object_detection/src/pose_detection.rs
@@ -133,7 +133,7 @@ impl PoseDetection {
         {
             let earlier = SystemTime::now();
 
-            load_into_scratchpad(tensor.get_data_mut::<f32>()?, image);
+            load_into_scratchpad(tensor.get_data_mut()?, image);
 
             context.preprocess_duration.fill_if_subscribed(|| {
                 SystemTime::now()

